lib/ssh_scan/rule_engine.rb

```diff
@@ -0,0 +1,94 @@
+module SSHScan
+  class RuleEngine
+    # Simple one rule matching.
+    def self.matchSingleRule(key, value, result)
```

Comment:
Maybe we should start using YARD documentation for these interfaces, so it's more clear what each parameter class type is?